source/collections/VectorList.ooc

```diff
@@ -121,9 +121,14 @@ VectorList: class <T> {
 	}
 	getSlice: func (range: Range) -> This<T> {
 		result := This<T> new(range count)
-		result _count = range count
-		source := (this pointer as Long + (range min * (T size))) as Pointer
-		memcpy(result pointer, source, range count * (T size))
+		this getSliceInto(range, result&)
 		result
 	}
+	getSliceInto: func (range: Range, buffer: This<T>@) {
+		if (buffer _vector capacity < range count)
+			buffer _vector resize(range count)
+		buffer _count = range count
+		source := (this pointer as Long + (range min * (T size))) as Pointer
```

Comment:
Actually shouldn't this cast use *(unsigned) long long* to be sure that we won't cast to a smaller type on 64 bit systems ? sizeof(long) is not guaranteed to be 8 bytes.
I assumed Long to be 8 bytes, but I'm not sure anymore. I don't understand this code:
```ooc
Long:  cover from signed long  extends LLong
```
does it mean that Long has the size of *signed long* (because it is a *cover* from this type) or that Long has the size of *LLong* (because it *extends* this type) ?